### ISSUE TITLE
Types non inferred with Cython modules

### ISSUE THREAD

**Issue by ubitux** [microsoft/pyright#3432]
**Describe the bug**
Check for symbols in a Cython module are not made while they are for a Python module.

Reproduced with `pyright 1.1.244`

**To Reproduce**
Given the follow tree/files:
```
.
├── my_cython_module
│   ├── my_cython_module.pyx
│   └── setup.py
└── tests
    └── my_cython_test.py
```

`my_cython_module/my_cython_module.pyx`:
```python
class MyCythonClass:
    def my_method(self):
        print("calling my Cython method")
```

`my_cython_module/setup.py`:
```python
from setuptools import setup
from Cython.Build import cythonize

setup(
    ext_modules = cythonize("my_cython_module.pyx")
)
```

`tests/my_cython_test.py`:
```python
import my_cython_module as mcm

c = mcm.MyCythonClass()
c.my_method()

c = mcm.MyNonExistingCythonClass()
c.my_non_existing_method()
```

Running `pyright` after setup:
```
$ python -m venv venv
$ . venv/bin/activate
$ pip install cython
[...]
$ pip install ./my_cython_module
[...]
$ pyright --lib tests/my_cython_test.py
No configuration file found.
No pyproject.toml file found.
stubPath /home/ux/sandbox/pyright/cython-test/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
0 errors, 0 warnings, 0 informations
Completed in 0.62sec
```

**Expected behavior**
I Would expect the following: `error: "MyNonExistingCythonClass" is not a known member of module (reportGeneralTypeIssues)` just like it does when the module is a native Python module (see #3429 for the Python equivalent code, which raises the error after being explicit with `--lib`, which does not have any effect here).

**Comment by erictraut** [comment]
The current behavior is correct. Pyright cannot infer types from a native module. Doing so would require it to load and execute the code in that module, which would be a security issue. As a static analysis tool, pyright never executes any of the code it's analyzing. It treats all symbols imported from native libraries (and by default, all symbols imported from untyped libraries) as `Unknown`. If you enable the `reportMissingTypeStubs` configuration option, pyright will emit an error for these cases.

**Comment by ubitux** [comment]
Adding `# pyright: reportMissingTypeStubs=true` at the top of `tests/my_cython_test.py` and running `pyright --lib tests/my_cython_test.py` does not seem to have any effect though.
